inferno-core/test/Eval/Spec.hs

```diff
@@ -281,10 +284,19 @@ evalTests = describe "evaluate" $
     shouldEvaluateTo "fromOption 0.0 None" $ VDouble 0
     shouldEvaluateTo "(Some 4.0) ? 0" $ VDouble 4
     shouldEvaluateTo "None ? 0.0" $ VDouble 0
+    shouldEvaluateTo "(Some 4.0) ? None ? 0" $ VDouble 4
+    shouldEvaluateTo "None ? (Some 4.0) ? 0" $ VDouble 4
+    shouldEvaluateTo "None ? None ? 4.0" $ VDouble 4
+    -- shouldEvaluateTo "(Some 4.0) <|> None <|> None" $ VOne $ VDouble 4
+    -- shouldEvaluateTo "None <|> (Some 4.0) <|> Some 3" $ VOne $ VDouble 4
+    -- shouldEvaluateTo "None <|> None <|> Some 4.0" $ VOne $ VDouble 4
     shouldEvaluateTo "Option.reduce (fun d -> d + 2) 0.0 (Some 4)" $ VDouble 6
     shouldEvaluateTo "Option.reduce (fun d -> d + 2) 0.0 (Some 4.0)" $ VDouble 6
     shouldEvaluateTo "Option.reduce (fun d -> d + 2) 0 (Some 4.0)" $ VDouble 6
     shouldEvaluateTo "Option.reduce (fun d -> d + 2) 0.0 None" $ VDouble 0
+    shouldEvaluateTo "Option.join None" $ VEmpty
+    shouldEvaluateTo "Option.join (Some None)" $ VEmpty
+    shouldEvaluateTo "Option.join (Some (Some 2.3))" $ VOne $ VDouble 2.3
```

Comment:
maybe add this one too:
```
shouldEvaluateTo "Option.join (Some 2.3)" $ VOne $ VDouble 2.3
```

Reply:
Aha, I added that but the test failed! `Option.join : option of (option of 'a) -> option of 'a`, so it can't be applied to `Some 2.3 : option of double`. Glad we have tests :)